File: forgemod/src/main/java/kz/uwu/komarubrowser/KomaruBrowser.java
package kz.uwu.komarubrowser;

import com.mojang.logging.LogUtils;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.server.ServerStartingEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.config.ModConfig;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import org.slf4j.Logger;

// The value here should match an entry in the META-INF/mods.toml file
@Mod(KomaruBrowser.MODID)
public class KomaruBrowser {
  // Define mod id in a common place for everything to reference
  public static final String MODID = "komarubrowser";
  // Directly reference a slf4j logger
  private static final Logger LOGGER = LogUtils.getLogger();

  public KomaruBrowser(FMLJavaModLoadingContext context) {

    // Register ourselves for server and other game events we are interested in
    MinecraftForge.EVENT_BUS.register(this);

    // Register our mod's ForgeConfigSpec so that Forge can create and load the
    // config file for us
    context.registerConfig(ModConfig.Type.SERVER, Config.SPEC);
  }

  // You can use SubscribeEvent and let the Event Bus discover methods to call
  @SubscribeEvent
  public void onServerStarting(ServerStartingEvent event) {
    // Do something when the server starts
    LOGGER.info("HELLO from server starting");
  }
}

File: forgemod/src/main/kotlin/kz/uwu/komarubrowser/KomaruBrowser.kt
package kz.uwu.komarubrowser

import net.minecraft.client.Minecraft
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent
import net.minecraftforge.fml.event.lifecycle.FMLDedicatedServerSetupEvent
import org.apache.logging.log4j.Level
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import thedarkcolour.kotlinforforge.forge.runForDist


// The value here should match an entry in the META-INF/mods.toml file
@Mod(KomaruBrowser.MODID)
object KomaruBrowser {
  const val MODID = "komarubrowser";

  // the logger for our mod
  val LOGGER: Logger = LogManager.getLogger()


  init {
    LOGGER.log(Level.INFO, "Hello world!")

    MOD_BUS.addListener(::onServerSetup)
  }

  /**
   * Fired on the global Forge bus.
   */
  private fun onServerSetup(@Suppress("UNUSED_PARAMETER") event: FMLDedicatedServerSetupEvent) {
    LOGGER.log(Level.INFO, "Server starting...")
  }
}
